core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -429,6 +455,46 @@ angular.module('oppia').factory('TopicEditorStateService', [
        */
       isSavingTopic: function() {
         return _topicIsBeingSaved;
+      },
+
+      /**
+       * Returns the classroom name for the topic.
+       */
+      getClassroomUrlFragment: function() {
+        return _classroomUrlFragment;
+      },
+
+      changeTopicWithNameExists: function(topicName, successCallback) {
```

Comment:
Hey @kevintab95 Should this be ```checkTopicWithNameExists``` ?

Reply:
Hm, it actually updated the variable "topicWithNameExists" so change seemed more appropriate.

Reply:
But semantically the wording doesn't make sense, and ditto for setTopicWithNameExists, etc. -- that doesn't really make sense as a clause in English, so it leads to confusion. I think what you really want to say is "change the _topicWithUrlFragmentExists property" but it reads as "change the topic, under certain conditions". So @Showtim3 has a point here.

It's hard to find a good name. How about doesTopicWithNameExistAsync()? The fact that it updates an inner "caching" variable behind the scenes seems fine, other functions do this too and don't call it out. I would also just drop the setter functions for these "inner" variables and inline that functionality.

One other note is that you should probably design this so that it updates a variable which is used directly (or via a getter) as the source of truth for the template, or return a value that can be used as the source of truth, but not both. The former is probably better, in which case you could call this function checkExistenceOfTopicName() and not have it have a callback. Do you think that would be possible?

Reply:
Notes from discussion with @seanlip :

With the current implementation "changeTopicWithNameExists" does not _return_ anything. It does 2 things:

1. Sets _topicWithNameExists. It does so by making a backend call that returns a boolean value. This boolen value is used to set _topicWithNameExists.
2. Executes the success callback. It does not pass any value to the success callback -- the callback being executed just indicates that _topicWithNameExists has been updated.

In order to get the latest value of _topicWithNameExists, the corresponding getter method "getTopicWithNameExists" should be used. Therefore, getTopicWithNameExists is always the source of truth.

That said, I will be changing the names used to avoid confusion and add jsdoc for the same.

Proposed name:

`changeTopicWithNameExists` -> `updateExistenceOfTopicName`